src/java.base/share/classes/java/util/stream/Collectors.java

```diff
@@ -1972,6 +1972,35 @@ R get() {
             this.forFalse = forFalse;
         }
 
+        @Override
+        public int size() {
+            return 2;
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return false;
+        }
+
+        @Override
+        public T get(Object key) {
+            if (key instanceof Boolean b) {
+                return b.booleanValue() ? forTrue : forFalse;
```

Comment:
Usages of partitioningBy are likely to be followed by get(TRUE) and get(FALSE) to get the sets in each partition, so this looks good, avoids needing the entrySet and and iterator.